addons/pos_loyalty/models/pos_config.py

```diff
@@ -10,30 +10,66 @@ class PosConfig(models.Model):
     def _default_loyalty_program(self):
         return self.env['loyalty.program'].search([('program_type', '=', 'loyalty')], limit=1)
 
-    use_coupon_programs = fields.Boolean('Coupons & Promotions',
-        help="Use coupon and promotion programs in this PoS configuration.")
-    coupon_program_ids = fields.Many2many(
-        'loyalty.program', string="Coupon Programs", domain=[('program_type', '=', 'coupons')],
-        relation='pos_config_coupon_program_rel')
-    promo_program_ids = fields.Many2many(
-        'loyalty.program', string="Promotion Programs", domain=[('program_type', '=', 'promotion')],
-        relation='pos_config_promo_program_rel')
-
-    loyalty_program_id = fields.Many2one('loyalty.program', "PoS Loyalty Programs", domain=[('program_type', '=', 'loyalty')], default=_default_loyalty_program)
-
-    use_gift_card = fields.Boolean('Gift Cards')
-    gift_card_program_id = fields.Many2one('loyalty.program', "PoS Gift Card Program", domain=[('program_type', '=', 'gift_card')])
+    use_coupon_programs = fields.Boolean('Coupons & Promotions', help="Use coupon and promotion programs in this PoS configuration.",
+        compute="_compute_use_coupon_programs")
+    coupon_program_ids = fields.Many2many('loyalty.program', string="Coupon Programs", domain=[('program_type', '=', 'coupons')], 
+        relation='pos_config_coupon_program_rel', compute="_compute_coupon_program_ids")
+    promo_program_ids = fields.Many2many('loyalty.program', string="Promotion Programs", domain=[('program_type', '=', 'promotion')], 
+        relation='pos_config_promo_program_rel', compute="_compute_promo_program_ids")
+
+    loyalty_program_id = fields.Many2one('loyalty.program', "PoS Loyalty Programs", domain=[('program_type', '=', 'loyalty')],
+        compute="_compute_loyalty_program_id")
+
+    use_gift_card = fields.Boolean('Gift Cards', compute="_compute_use_gift_card")
+    gift_card_program_id = fields.Many2one('loyalty.program', "PoS Gift Card Program", domain=[('program_type', '=', 'gift_card')],
+        compute="_compute_gift_card_program_id")
     gift_card_settings = fields.Selection(
         [
-            ("create_set", "Generate a new barcode and set a price"),
-            ("scan_set", "Scan an existing barcode and set a price"),
-            ("scan_use", "Scan an existing barcode with an existing price"),
+            ("create_set", "Generate PDF cards"),
+            ("scan_use", "Scan existing cards"),
         ],
         string="Gift Cards settings",
         default="create_set",
         help="Defines the way you want to set your gift cards.",
     )
 
+    @api.depends('all_program_ids')
```

Comment:
As all computes have the same depends, you can do only one compute that fill all the fields and use filtered, something like:

@api.depends('all_program_ids')
    def _compute_program_ids(self):
        for config in self:
            config.loyalty_program_ids = config.all_program_ids.filtered(lambda p: p.program_type == 'loyalty')
            config.coupon_program_ids = config.all_program_ids.filtered(lambda p: p.program_type == 'coupon')
            .....
            config.use_coupon_programs = bool(config.coupon_program_ids or config.promo_program_ids)